autoload/ale/definition.vim

```diff
@@ -17,6 +20,18 @@ function! ale#definition#ClearLSPData() abort
     let s:go_to_definition_map = {}
 endfunction
 
+function! ale#definition#UpdateTagStack() abort
+    let l:should_update_tagstack = exists('*gettagstack') && exists('*settagstack') && g:ale_update_tagstack
+
+    if l:should_update_tagstack
+        let l:from = [bufnr('%'), line('.'), col('.'), 0]
+        let l:tagname = expand('<cword>')
+        let l:winid = win_getid()
```

Comment:
Get the window ID from the buffer number, instead of the current window.

Reply:
This sounds like it's probably *really* relevant and important - but I'm afraid I'm not an expert on the Vim functions involving windows and buffers.  Any chance you can provide some pointers to the relevant docs?